### ISSUE TITLE
Remove bundled ancient samtools 0.1.19 and move to building against upstream htslib

### ISSUE THREAD

**Issue by SoapZA** [cbg-ethz/shorah#47]
This is much easier and shifts the complexity away from ShoRAH.

**Comment by SoapZA** [comment]
@DrYak any movement here? I would want this in before release 2.0.0

**Comment by DrYak** [comment]
Working on it, that's my next biggest point on my todo list.
I think I should be able to have something by the end of the next week.
Would that be okay for your timing ?

**Comment by SoapZA** [comment]
@DrYak how far are we here?

**Comment by DrYak** [comment]
Sorry got distracted by [some problems the cancer-team neighbors were having](https://github.com/DrYak/louvain-community/commit/cb22cfbfd7cda0dd3138fcd6197a609bab72d985#diff-40ab9c3fe6e1c7afbcb549a3ecdf5bf0R55)

I'm working on it. 
Getting a hang on how pileup have changed between versions (the API [that ShoRAH](https://github.com/cbg-ethz/shorah/blob/f7c38a46c94078a086f94a51bffc85d5c3d25d74/src/cpp/b2w.cpp#L220) has been using has been moved out of the library and [into samtools executables themselves](https://github.com/samtools/samtools/blob/618d6244b62125b44c828284eee441780d193349/bam_plbuf.h#L46) )

**Comment by DrYak** [comment]
(Sorry for slow updates, was sick last week)

B2W ported to HTSLib and produces identical results for the few test cases I have laying around.

Bascially :
 - I have moved the common parts between "scanning all chromosome" - "scanning a single r.o.i." into a single lamba.
- `bam_fetch` has been moved from the library into samtools. I've re-implemented it in b2w.c based on samtool/bam.c, as recommended by samtools authors.
- as there is no call-backs involved anymore (`bam_fetch` is simply a while loop), i've moved the `fetch_func1` and `fetch_func2` into the while body (each was called exactly once, anyway)
- `bam_plbuf_push`  has been moved from the library into samtools. I've re-implemented it in b2w.c based on samtool/bam_plbuf.c, as recommended by samtools authors.
- as there are no call-backs involved anymore (`bam_plbuf_push` too is simply a while loop), I've slightly modified `pile_func` and call it directly from each while loop.
- command line parameters are stored in a separate struct `param`.
- work data is accessed through the lambda's capture, instead of being passed around through a struct.

Now moving onto pushing the changes to FIL, too.

**Comment by DrYak** [comment]
And FIL port to HTSLib has been pushed to the [with_htslib](https://github.com/cbg-ethz/shorah/tree/with_htslib) branch.

Basically the same mods :
 - the loops for amplicon mode and normal mode have been fused (as we are only adding columns to the input SNV, not touching frequencies or priors).
 - `bam_fetch` and `bam_plbuf_push` re-implemented based on samtools, as recommended by HTSlib authors.

**Comment by DrYak** [comment]
Updated Travis CI yaml file to fetch HTSLib (from Debian Med to have a not to ancient version).
Updated configure.ac to request C++11 (the antediluvian GCC 4.8.3 featured on Trust Thar used by defaults to older dialects).

Now Travis tests successfully pass.

Speaking of tests, I'm also able to replicate the result I'm getting from pre-HTSLib FIL and B2W.

I'll ask Susie for datafiles for which she was receiving bad/weird results that she suspected were due to bugs from a way to old samtools lib version.

I'm done on my side, I'll need your knowledge of Meson to make the same change as I did to the autotools/m4 side.
